src/diagnostics.rs

```diff
@@ -52,7 +52,7 @@ pub fn publish_diagnostics(params: Params, ctx: &mut Context) {
         .iter()
         .map(|x| {
             format!(
-                "{}|{}",
+                "'{}|{}'",
```

Comment:
> When there are no errors, you set `lsp_error_lines` to `0 '0| '`

Good catch; I've added another commit that simply uses `%opt{lsp_diagnostic_line_error_sign}` as the default width.

I think that's good enough for now because users probably want to configure all 4 symbols to the same width, to avoid the same glitch.

It's a bit hacky, but so is this quoting -- it means that `'` needs to be doubled if users want that in the gutter.
Neither is a problem, so I'd merge like this and perhaps in future implement this on the Rust side.

> I use nerd-font icons as my symbols, but they are wider than an actual character and thus I need to add a whitespace character afterwards. Otherwise the icon overlaps the line-numbers

Sounds like Kakoune (which uses the system's wcwidth()) disagrees on the width of the character with your terminal.
This is typical of macOS systems because those ship old versions of wcwidth(), and as a result terminals opt to bundle newer ones..
We could also do that -- using something like https://github.com/ridiculousfish/widecharwidth -- but that would probably make things worse.